src/main/python/janusgraph_python/structure/io/GraphsonReader.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2018 JanusGraph Python Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from gremlin_python.structure.io.graphsonV3d0 import GraphSONReader
+from gremlin_python.structure.io.graphsonV3d0 import GraphSONUtil
+from ...serializer.RelationIdentifierDeserializer import RelationIdentifierDeserializer
+
+
+class JanusGraphSONReader(object):
+    """
+    This class registers JanusGraph-specific deserializers so that objects like GeoShape, RelationIdentifier
+    can be interpreted on Python client side.
+    """
+
+    GRAPHSON_PREFIX = "janusgraph"
+    GEO_GRAPHSON_BASE_TYPE = "Geoshape"
+    RELATIONID_BASE_TYPE = "RelationIdentifier"
+    GeoShape_GRAPHSON_TYPE = GraphSONUtil.formatType(GRAPHSON_PREFIX, GEO_GRAPHSON_BASE_TYPE)
```

Comment:
Please use consistent naming. Why all uppercase, but suddenly pascal case for _GeoShape_? Constants [should be named](https://www.python.org/dev/peps/pep-0008/#constants) with all capital letters, but this is strictly speaking not a constant as you compute the value when the object is initialized. So, it's an instance variable and that should be named lowercase.

I wonder however why you use 5 lines to define 3 constants and 2 instance variables with two method calls if you also could simply write this as:

```python
GEOSHAPE_GRAPHSON_TYPE = "janusgraph:Geoshape"
RELATIONID_GRAPHSON_TYPE = "janusgraph:RelationIdentifier"
```

or even simply hard code where you use it which should only be one place.